### ISSUE TITLE
Huobi: Cannot place an order in one-way mode on the USDT-M cross contract

### ISSUE THREAD

**Issue by bizantom** [ccxt/ccxt#15250]
- OS: MAC OS X
- Programming Language version: JS
- CCXT version: 1.95.36

Hi,

I don't manage to place an order on the USDT-M cross margin contract when the one-way mode is enabled. I got the following error: _One-way mode currently; Unavailable to place orders in hedge mode._ 
I don't have any opened position nor opened order, so I don't get what means this error message.

The same code is working if I am in hedge mode.

Do you have any idea why I get this error ?

```JavaScript
const client = new huobi({
	apiKey: apiKey,
	secret: apiSecret,
	options: {
		defaultType: 'future',
		defaultSubType: 'linear',
		marginMode: 'cross'
	}
});

this.client.verbose = true;
const order = await this.client.createOrder(
   'BTC/USDT:USDT',
   'optimal_20_fok',
   'buy',
   10,
   null,
   {
	offset: 'open',
	'client-order-id': 'an_order_id',
	channel_code: ''
   }
);
		
```

Here is the trace of the API call
```
fetch Request:
 huobi POST https://api.hbdm.com/linear-swap-api/v1/swap_cross_order?AccessKeyId=aad1aaf9-4314f653-32df7bae-rfhfg2mkl3&SignatureMethod=HmacSHA256&SignatureVersion=2&Timestamp=2022-10-11T18%3A18%3A34&Signature=W7ZyCXgzUj6XRK2uUkVqXXS8UX1EZxATxf0%2F0JQHDkQ%3D 
RequestHeaders:
 {
  'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/100.0.4896.75 Safari/537.36',
  'Content-Type': 'application/json'
} 
RequestBody:
 {"contract_code":"BTC-USDT","volume":"4","direction":"buy","offset":"open","lever_rate":1,"order_price_type":"optimal_20_fok","channel_code":"","client-order-id":"1376_1"} 

handleRestResponse:
 huobi POST https://api.hbdm.com/linear-swap-api/v1/swap_cross_order?AccessKeyId=............
ResponseHeaders:
 {
  'Cache-Control': 'no-cache, no-store',
  'Cf-Cache-Status': 'DYNAMIC',
  'Cf-Ray': '758999618bc4695d-FRA',
  Connection: 'close',
  'Content-Encoding': 'gzip',
  'Content-Type': 'text/plain;charset=UTF-8',
  Date: 'Tue, 11 Oct 2022 18:18:35 GMT',
  Expires: 'Thu, 01 Jan 1970 00:00:00 GMT',
  Pragma: 'no-cache',
  'Ratelimit-Interval': '3000',
  'Ratelimit-Limit': '72',
  'Ratelimit-Remaining': '71',
  'Ratelimit-Reset': '1665512318266',
  Reqid: '1bc89887546fd11f7108788e4d5dba2c',
  'Request-Id': '8e4e8b845ac43d4bb7ff3b0bbdf03119',
  Server: 'cloudflare',
  'Transfer-Encoding': 'chunked',
  'X-Request-Id': '1bc89887546fd11f7108788e4d5dba2c'
} 
ResponseBody:
 {"status":"error","err_code":1500,"err_msg":"One-way mode currently; Unavailable to place orders in hedge mode. ","ts":1665512315272} 
```

**Comment by samgermain** [comment]
There's a few problems with your code. 

- You're assigning the constant `client` and then calling `this.client`, `this.client` isn't a thing, take away the `this`
- You're setting defaultType of future even though the symbol you are reading is swap (read the [contract naming conventions](https://docs.ccxt.com/en/latest/manual.html#contract-naming-conventions))
- You're using await outside an async function

I updated your code to the code below and this works. I'm not sure why your got the `One-way mode currently; Unavailable to place orders in hedge mode`. Try the fixes I posted and reply if your have any more problems

```

const ccxt = require ('ccxt');
const keys = require('./keys.local.json')

const client = new ccxt.huobi({
	apiKey: keys.huobi.apiKey,
	secret: keys.huobi.secret,
	options: {
		defaultType: 'swap',
		defaultSubType: 'linear',
		marginMode: 'cross'
	}
});

// client.verbose = true;
async function main() {
	const order = await client.createOrder(
		'BTC/USDT:USDT',
		'optimal_20_fok',
		'buy',
		1,
		null,
		{
			offset: 'open',
			'client-order-id': 'an_order_id',
			channel_code: ''
		}
	);
	console.log (order);
};

main();
```

**Comment by bizantom** [comment]
Indeed I did not clean my code properly after extracting it from my project, sorry for that. But the issue is not here.

This request is rejected when we are in one-way mode but accepted when we are in hedge mode. I just got an answer from the Huobi support. The issue comes from the offset parameter. This one is required in hedge mode, but not in one-way mode. If it is set in one way-mode, the value should be 'both'.
The same example with 'both' instead of 'open' works well in one-way mode.
Just one note about the ccxt library, it forced me to set an _offset_ parameter, whereas in my case it was not mandatory. I think this should be changed.
